Added a cheatsheet to help operators when using the etcdctl tool

aaronfern opened the pull request:
**What this PR does / why we need it**:
This PR adds a cheatsheet that will help operators when debugging issues with etcd or using the `etcdctl` tool.
This script prints out file paths of certificates the `etcdctl` tool uses and also prints out some common `etcdctl` commands

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       breaking|feature|bugfix|doc|other
- target_group:   user|operator|developer|dependency
-->
```bugfix operator
Added a cheatsheet to help operators when debugging etcd
```

aaronfern commented:
Thanks for your review @petersutter 

I've changes the file permission and the command description, so I hope it runs now

> it would be nice if the user does not have to set the PATH env var and that it's part of the installation

I wanted this too initially, however I was unable to set the PATH env var from the script and hence I added a note for the user to manually do it. Any idea how I would be able to do that?

petersutter commented:
> it would be nice if the user does not have to set the PATH env var and that it's part of the installation
> 
> > I wanted this too initially, however I was unable to set the PATH env var from the script and hence I added a note for the user to manually do it. Any idea how I would be able to do that?

@plkokanov maybe?

petersutter commented on hacks/print-etcd-cheatsheet.sh:
```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
```
drop `.sh` suffix for this file, otherwise you can't just run `print-etcd-cheatsheet`

aaronfern replied:
When creating the image, [this line](https://github.com/gardener/ops-toolbelt/pull/100/files#diff-585e09338be8d61489c23232e5fe922dae7a54726fcbc6ae95e242750ecdbdcaR123) copies the content of this file to `print-etcd-cheatsheet`. 
Will this not be sufficient? It is then possible to run `print-etcd-cheatsheet` within the image

plkokanov replied:
There is a previous copy operation which copies all `./hacks` files to the `/hacks` dir on the container: https://github.com/gardener/ops-toolbelt/blob/f04ab43a645098a45eb6a8768d6e8130b9f45e52/dockerfile-configs/common-components.yaml#L102-L105
So this file will be available twice (this is also the case for `install_etcdctl`). However, it will be available with the `.sh` suffix in the `/hacks` directory.

aaronfern replied:
Fair enough 
dropped `.sh` suffix for `print-etcd-cheatsheet`

petersutter commented on hacks/install_etcdctl:
```diff
@@ -43,6 +43,7 @@ function install () {
   echo "You can now start using etcdctl. Just execute \"etcdctl\" to use it. See https://etcd.io/docs/v3.4/dev-guide/interacting_v3/ for more details."
   echo "This tool assumes that it is being run in an ephemeral container in a pod. If this is not the case then please ensure that you provide the correct ecrtificates, the correct endpoint and have all necessary accesses."
   echo "Certificates to be passed to the command should be mounted onto any container in the pod having a shared process namespace. Please run \"ps -A\" to list all processes and then access the certificates using \"/proc/<proc number>/root/<file-path>\". Pass these certificate file paths to the etcdctl command :)"
+  echo "Please run \"PATH=$PATH:$PWD\" to add this dir to the PATH env var to help you easily use the etcdctl tool"
```
```suggestion
  echo "Please run \"export PATH=$PATH:$PWD\" to add this dir to the PATH env var to help you easily use the etcdctl tool"
```

petersutter commented on dockerfile-configs/common-components.yaml:
```diff
@@ -117,6 +117,13 @@
+    from: ./hacks/print-etcd-cheatsheet.sh
+    to: /nonroot/hacks/print-etcd-cheatsheet
+    command: --chown=65532:65532
+    info: Bash script which prints out etcd certificate paths and handy etcdctl comands. If you are running this container as a non-root container, execute the script in the `/nonroot/hacks` directory. Else use the one in `/hacks`
```
in my case I got the following error when running the script:
error: cannot find the process id for etcd-wrapper. Please ensure it is running

how to ensure that it is running / what should I do? :)

aaronfern replied:
This needs a running etcd process
easiest way is to use one of our dev clusters and mount this image as an ephemeral image onto the etcd pod 
(eg: `k debug -it etcd-main-0 --image=< ops-toolbelt image >`)

petersutter replied:
I think it would be helpful to add this to the info, except this is obvious for everyone except me :D

aaronfern replied:
Updated info to mention this 
Hope this is okay

plkokanov commented:
> > it would be nice if the user does not have to set the PATH env var and that it's part of the installation
> > > I wanted this too initially, however I was unable to set the PATH env var from the script and hence I added a note for the user to manually do it. Any idea how I would be able to do that?
> 
> @plkokanov maybe?

Hmm, it should be possible with a bash command similar to this: https://github.com/gardener/ops-toolbelt/blob/0a6d6f3f78cf96233eee39067c0f318eb06cfb78/dockerfile-configs/common-components.yaml#L141-L144

This `generator` stuff seems to be asking for some refactoring btw 😅

plkokanov commented:
I was mistaken in my previous [comment](https://github.com/gardener/ops-toolbelt/pull/100#issuecomment-1896257056)

The correct way to do that would be to extend the environment variables when building the container: https://github.com/gardener/ops-toolbelt/blob/f04ab43a645098a45eb6a8768d6e8130b9f45e52/dockerfile-configs/common-components.yaml#L146-L148

You can just add
```
  - PATH=/nonroot/hacks:$PATH
```

@aaronfern btw do you plan to continue with this PR. I'm really sorry for replying so late, but was busy with other stuff and this slipped my mind (yet again)

aaronfern commented:
Hey @plkokanov 
Yes, I will continue with this PR
Sorry, got sidetracked and lost sight of this PR

Will resolve all comments asap

aaronfern commented:
Thanks @petersutter and @plkokanov for your reviews and I'm very sorry for the delay in addressing your comments

I've addressed all your comments. PTAL whenever possible

plkokanov commented on hacks/print-etcd-cheatsheet:
```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+# Copyright 2023 SAP SE or an SAP affiliate company
```
Change the license header to the REUSE one

aaronfern replied:
Updated

plkokanov reviewed (changes requested):
Generally looks good, just license information has to be updated.

plkokanov reviewed (approved):
/lgtm
